x-pack/packages/kbn-elastic-assistant/impl/assistant/use_conversation/index.tsx

```diff
@@ -51,299 +48,264 @@ interface CreateConversationProps {
   const {
     allSystemPrompts,
     assistantTelemetry,
     knowledgeBase: { isEnabledKnowledgeBase, isEnabledRAGAlerts },
-    setConversations,
+    http,
   } = useAssistantContext();
 
+  const getConversation = useCallback(
+    async (conversationId: string) => {
+      const currentConversation = await getConversationById({ http, id: conversationId });
+      if (isHttpFetchError(currentConversation)) {
+        return;
+      }
+      return currentConversation;
+    },
+    [http]
+  );
+
   /**
    * Removes the last message of conversation[] for a given conversationId
    */
   const removeLastMessage = useCallback(
-    (conversationId: string) => {
+    async (conversationId: string) => {
       let messages: Message[] = [];
-      setConversations((prev: Record<string, Conversation>) => {
-        const prevConversation: Conversation | undefined = prev[conversationId];
-
-        if (prevConversation != null) {
-          messages = prevConversation.messages.slice(0, prevConversation.messages.length - 1);
-          const newConversation = {
-            ...prevConversation,
-            messages,
-          };
-          return {
-            ...prev,
-            [conversationId]: newConversation,
-          };
-        } else {
-          return prev;
-        }
-      });
+      const prevConversation = await getConversationById({ http, id: conversationId });
+      if (isHttpFetchError(prevConversation)) {
+        return;
+      }
+      if (prevConversation != null) {
+        messages = prevConversation.messages.slice(0, prevConversation.messages.length - 1);
+        await updateConversationApi({
+          http,
+          conversationId,
+          messages,
+        });
+      }
       return messages;
     },
-    [setConversations]
+    [http]
   );
 
   /**
    * Updates the last message of conversation[] for a given conversationId with provided content
    */
   const amendMessage = useCallback(
-    ({ conversationId, content }: AmendMessageProps) => {
-      setConversations((prev: Record<string, Conversation>) => {
-        const prevConversation: Conversation | undefined = prev[conversationId];
-
-        if (prevConversation != null) {
-          const { messages, ...rest } = prevConversation;
-          const message = messages[messages.length - 1];
-          const updatedMessages = message
-            ? [...messages.slice(0, -1), { ...message, content }]
-            : [...messages];
-          const newConversation = {
-            ...rest,
-            messages: updatedMessages,
-          };
-          return {
-            ...prev,
-            [conversationId]: newConversation,
-          };
-        } else {
-          return prev;
-        }
-      });
+    async ({ conversationId, content }: AmendMessageProps) => {
+      const prevConversation = await getConversationById({ http, id: conversationId });
+      if (isHttpFetchError(prevConversation)) {
+        return;
```

Comment:
Consider displaying an error toast indicating the last message in the conversation could not be updated